### ISSUE TITLE
Wildcard support for file extensions?

### ISSUE THREAD

**Issue by NicolasReibnitz** [loeffel-io/ls-lint#56]
Hi!

First of all, I'd like to say that I love this! Exactly what I was looking for! Thanks!

One feature that seems to be missing (or would be great to have) is the option to set a general rule for all files. Specifically to say that all files and folders should be lowercase, regardless of their extension.

It seems that something like this isn't possible:

```yml
ls:
    .*: lowercase
    .dir: kebab-case
    .js: kebab-case
    ...
```

Correct?

The alternative would be to find out what extensions are in a project and what extensions could come up in the future, which isn't really feasible.

Or do you have another idea?

Thanks in advance!

**Comment by jgornick** [comment]
I've found myself wanting to use wildcards (i.e. glob) for other things like ignore paths as well.

Is it possible to use globs wherever matching needs to happen (e.g., extensions, paths, ignore)?

**Comment by aphecetche** [comment]
Indeed, using a default rule for all files (possibly in a subdirectory) would be super helpful. My use case : checking that image files are all kebab case. Currently I cannot use `ls-lint` for that as for instance I don't know how to write a rule for `don_t_like_this_name.JPG` because it's the _extension_ that is breaking the naming convention...

**Comment by loeffel-io** [comment]
Hey, 

agree to all of you and i'll try to bring more and more glob patterns to ls-lint.

@aphecetche `.JPG: kebeb-case` isn't working?

**Comment by aphecetche** [comment]
> Hey,
> 
> agree to all of you and i'll try to bring more and more glob patterns to ls-lint.
> 
> @aphecetche `.JPG: kebeb-case` isn't working?

sorry for the late answer : no, it does not work (at least for what I want to do, I.e. have the full filename be kebab-case), as the casing of the extension itself is not checked as far as I can tell.

**Comment by crazyx13th** [comment]
Wildcard support would be perfect (but not only for ignores)
I plan to define a regexp to forbid some file-extensions
```yml
ls:
  
  components/assets:
    .dir: kebab-case
    .png: kebab-case
    .*: regex: only-png-files-allowed
```

**Comment by tigerinus** [comment]
wildcard support, or default rule, please

**Comment by Igorkowalski94** [comment]
As an alternative, you can use [eslint-plugin-project-structure](https://github.com/Igorkowalski94/eslint-plugin-project-structure), you will have eslint errors in real time.

features:
✅ Validation of project structure (Any files/folders outside the structure will be considered an error).
✅ Validation of folder and file names.
✅ Name case validation.
✅ Name regex validation.
✅ File extension validation (Support for all extensions).
✅ Inheriting the parent's name (The child inherits the name of the folder in which it is located).
✅ Folder recursion (You can nest a given folder structure recursively).
✅ Forcing a nested/flat structure for a given folder.

**Comment by ethanjdiamond** [comment]
Love the library! Thanks for making it.

I'm going to add another use case for this that I've run into, which is sub-extensions. My app uses SvelteKit and Vite, which means I'm dealing with .d.ts, .config.ts, .server.ts. For all of those I want camelCase for the base name and would love to be able to do `.*.ts: camelcase` instead of individually calling out all those sub-extensions.

**Comment by ethanjdiamond** [comment]
@loeffel-io Honestly, neither. I would have expected it to go top to bottom, and check against the first match. So if `.service.*` was first in the list under `ls:`, I'd expect the file to be checked for camelCase, and if you moved `.*.gif` to be the first rule, I'd expect the file to be checked for PascalCase.

**Comment by loeffel-io** [comment]
@ethanjdiamond, thank you for the contribution 🙏  i think this would kill some of the simplicity ls-lint is driven for. the yaml file is is map data structure which is firstly unordered - i will think about this

**Comment by loeffel-io** [comment]
This feature is now available in the `v2.3.0-beta.1` release. 
Please test it and let me know if there are any issues ❤️ 

The upcoming v2.3.0 blog post: https://ls-lint.org/blog/announcements/v2.3.0.html
The docs: https://ls-lint.org/2.3/getting-started/introduction.html
The release notes: https://github.com/loeffel-io/ls-lint/releases/tag/v2.3.0-beta.1
